contracts/ETHRegistrarController.sol

```diff
@@ -58,18 +64,24 @@ contract ETHRegistrarController is Ownable {
     }
 
     function makeCommitment(string memory name, address owner, bytes32 secret) pure public returns(bytes32) {
+        return makeCommitmentWithConfig(name, owner, secret, address(0), address(0));
+    }
+
+    function makeCommitmentWithConfig(string memory name, address owner, bytes32 secret, address resolver, address addr) pure public returns(bytes32) {
         bytes32 label = keccak256(bytes(name));
-        return keccak256(abi.encodePacked(label, owner, secret));
+        if (resolver == address(0) && addr == address(0)) {
+            return keccak256(abi.encodePacked(label, owner, secret));
+        }
+        return keccak256(abi.encodePacked(label, owner, secret, resolver, addr));
     }
 
     function commit(bytes32 commitment) public {
         require(commitments[commitment] + maxCommitmentAge < now);
         commitments[commitment] = now;
     }
 
-    function register(string calldata name, address owner, uint duration, bytes32 secret) external payable {
+    function _consumeCommitment(string memory name, uint duration, bytes32 commitment) internal returns (uint256) {
```

Comment:
function ordering in contracts is something I usually like doing
- external
- public
- private
- internal

Reply:
I've moved `_consumeCommitment` (the new internal function) to the end of the contract.

I have not re-ordered the existing functions though. Let me know if I should. There are also several functions which I think could be changed from `public` to `external`; if you want the functions re-ordered, I could make these changes too.

Let me know. :)